otter/test/test_controller.py

```diff
@@ -1464,3 +1471,96 @@ def test_checks_pass_replace_true_no_purge_failure(self):
         ])
         self.assertRaises(ValueError, self._remove, True, False,
                           seq_dispatcher)
+
+    def test_perform_convergence_remove_from_group(self):
+        """
+        Perform :func:`convergence_remove_server_from_group` with the given
+        dispatcher.
+        """
+        self.state.desired = 2
+        fake_super = FakeSupervisor()
+
+        dispatcher = ComposedDispatcher([
+            test_dispatcher(),
+            SequenceDispatcher([
+                self._tenant_retry(
+                    get_server_details('server_id').intent,
+                    lambda _: (StubResponse(200, {}), self.server_details)),
+                (GetScalingGroupInfo(tenant_id='tenant_id',
+                                     group_id='group_id'),
+                    lambda _: (self.group, self.group_manifest_info)),
+                self._tenant_retry(
+                    mock.ANY,
```

Comment:
this mock.ANY is too lax I think, we should specifically be asserting that it's an `EvictServerFromScalingGroup` with all the right parameters.